backends/xnnpack/runtime/XNNPACKBackend.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+#include <executorch/runtime/platform/compiler.h>
+
+namespace executorch::backends::xnnpack {
+ET_EXPERIMENTAL void set_workspace_sharing_enabled(bool enable);
```

Comment:
So from priority, compile_time > runtime? And at runtime, it's one shot at the start up? What about some thread turning it on vs. other off? I guess it's on them?

Reply:
I've updated for backend options. To maintain the current behavior, I'm setting global mode if ENABLE_XNNPACK_SHARED_WORKSPACE is set, though we probably want to re-evaluate this in the future, as per-model might be better.

The runtime backend option is global, but I've written the code in such way that we can set it per-method/model if/when backend options support this.